src/api/app/models/bs_request_action.rb

```diff
@@ -287,8 +287,14 @@ def notify_params(ret = {})
   end
 
   def contains_change?
-    return sourcediff.present?
-  rescue BsRequestAction::DiffError
+    source_info_target_pkg = Backend::Api::Sources::Package.source_info(target_project, target_package, nofilename: 1)
+    source_info_source_pkg = Backend::Api::Sources::Package.source_info(source_project, source_package, nofilename: 1)
+
+    hash_sum_target = Xmlhash.parse(source_info_target_pkg)['verifymd5']
+    hash_sum_source = Xmlhash.parse(source_info_source_pkg)['verifymd5']
+
+    hash_sum_target != hash_sum_source
```

Comment:
So there is no test for it :thinking: (or did it fail in travis and we didn't recognize?)

Reply:
I was already writing one. Probably finished this afternoon